controllers/csiaddonsnode_controller.go

```diff
@@ -58,5 +117,50 @@ func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 func (r *CSIAddonsNodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&csiaddonsv1alpha1.CSIAddonsNode{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// addFinalizer adds finalizer to object instance if it is not present.
+func (r *CSIAddonsNodeReconciler) addFinalizer(
+	ctx context.Context,
+	instance *csiaddonsv1alpha1.CSIAddonsNode,
+	logger *logr.Logger) error {
```

Comment:
context and logger are common arguments for functions of CSIAddonsNodeReconciler, use those as 1st parameters in the list. Arguments that change depending on the function (like CSIAddonsNode), pass those afterwards.

I'm not very fond of a name like `instance` or `object`. Better use `node` or something that is a little more descriptive.